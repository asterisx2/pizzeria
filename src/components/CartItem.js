import React from 'react';

const getDescription = (toppings) => {
  let desc = toppings.reduce((str, top) => str += top.name + ', ', '');
  return desc.substring(0, desc.length - 2);
}

const capitalizeFirstLetter = (string) => {
  return string.charAt(0).toUpperCase() + string.slice(1);
}

const calculateTotal = (pizza) => {
  return pizza.quantity * (pizza.basePrice + pizza.doughType.price + pizza.toppings.reduce((sum, top) => sum += top.price, 0));
}

export default ({ pizza, removePizza }) => {
  return (
    <div 
      className="d-flex justify-content-between pt-2 pb-2">
      <div 
        className="d-flex">
        <img 
          src={pizza.imgPath} 
          className="fluid-img-large" 
          alt="" />
        <div 
          className="d-flex flex-column text-left pl-2">
          <span 
            className="font-small color-orange">
            Size: {capitalizeFirstLetter(pizza.size)}
          </span>
          <span 
            className="font-small color-orange">
            Dough: {pizza.doughType.name}
          </span>
          <span 
            className="text-primary font-italic font-weight-light font-small">
            {getDescription(pizza.toppings)}
          </span>
        </div>
      </div>
      <div 
        className="d-flex flex-column text-right justify-content-between">
        <i 
          className="fa fa-times-circle cursor-pointer text-danger mt-2" 
          aria-hidden="true" 
          onClick={removePizza}></i>
        <span 
          className="font-small">
          &euro;{calculateTotal(pizza).toFixed(2)}
        </span>
      </div>
    </div>
  )
}