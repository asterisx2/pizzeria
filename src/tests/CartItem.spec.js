import React from 'react';
import CartItem from '../components/CartItem';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe('Cart Component', () => {
  it('renders without crashing', () => {
    let pizza = {
      id: 1,
      size: 'small',
      basePrice: 199,
      quantity: 1,
      toppings: [
        {
          __typename: 'topping',
          name: 'pepperoni',
          price: 40
        }
      ],
      doughType: {
        id: 1,
        name: 'Thin',
        description: 'Thin and cristy Pizza dough',
        price: 0
      },
      imgPath: ''
    };

    const tree = renderer
      .create(
        <MemoryRouter>
          <CartItem pizza={pizza} removePizza={() => { }} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // TODO
  it('handles remove pizza correctly', () => { });
});
