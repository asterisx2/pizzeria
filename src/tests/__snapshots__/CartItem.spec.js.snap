// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Component renders without crashing 1`] = `
<div
  className="d-flex justify-content-between pt-2 pb-2"
>
  <div
    className="d-flex"
  >
    <img
      alt=""
      className="fluid-img-large"
      src=""
    />
    <div
      className="d-flex flex-column text-left pl-2"
    >
      <span
        className="font-small color-orange"
      >
        Size: 
        Small
      </span>
      <span
        className="font-small color-orange"
      >
        Dough: 
        Thin
      </span>
      <span
        className="text-primary font-italic font-weight-light font-small"
      >
        pepperoni
      </span>
    </div>
  </div>
  <div
    className="d-flex flex-column text-right justify-content-between"
  >
    <i
      aria-hidden="true"
      className="fa fa-times-circle cursor-pointer text-danger mt-2"
      onClick={[Function]}
    />
    <span
      className="font-small"
    >
      â‚¬
      239.00
    </span>
  </div>
</div>
`;
